components/script/dom/htmlformelement.rs

```diff
@@ -465,6 +466,15 @@ impl HTMLFormElement {
     /// https://html.spec.whatwg.org/multipage/#statically-validate-the-constraints
     fn static_validation(&self) -> Result<(), Vec<FormSubmittableElement>> {
         let node = self.upcast::<Node>();
+        for child in node.traverse_preorder() {
+            match child.downcast::<Element>() {
```

Comment:
```rust
if let Some(el) = child.downcast::<Element>() {
    if el.disabled_state() {
        continue;
    }
    let validatable = match el.as_maybe_validatable() {
        Some(v) => v,
        None => continue,
    };
    if !validatable.is_instance_validatable() {
        continue;
    }
    // invoke some other method that returns true if the form element validates, false otherwise.
}
```